pkg/deploy/deploy.go

```diff
@@ -175,7 +180,31 @@ func (d *deployer) Deploy(ctx context.Context) error {
 		}
 	}
 
-	return nil
+	return d.saveRPVersion()
```

Comment:
should be the last step at the end of deployer.Upgrade() - i.e. only update the blob version when we know for sure that there are no RPs at older versions still serving

Reply:
Done, and I included your explanation as a comment to help guide future development https://github.com/Azure/ARO-RP/pull/1033/commits/ca5dc0e667470c522f87b424e5d21706b120e349#diff-3365a9a01fd226a479859834fc467d90R44-R45